From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Telesphoreo <me@telesphoreo.me>
Date: Mon, 19 Oct 2026 22:26:09 -0400
Subject: [PATCH] Provide registry access for real item codec regression tests

Bind the existing dummy server to the integration test registry. Paper item component sanitation reads this registry during initialization; no Minecraft server is started.

diff --git a/src/test/java/org/bukkit/support/DummyServerHelper.java b/src/test/java/org/bukkit/support/DummyServerHelper.java
index 8d525380ed2c933431ec178204874f3a531d9e02..e6b7560e2f6ad68d479cd7da6563268d52667996 100644
--- a/src/test/java/org/bukkit/support/DummyServerHelper.java
+++ b/src/test/java/org/bukkit/support/DummyServerHelper.java
@@ -26,6 +26,19 @@ import org.bukkit.craftbukkit.util.Versioning;
 public final class DummyServerHelper {
 
     public static Server setup() {
+        // Scissors - network item codecs need the same registry access as the test environment.
+        if (net.minecraft.server.MinecraftServer.getServer() == null) {
+            net.minecraft.server.MinecraftServer minecraftServer = mock(net.minecraft.server.MinecraftServer.class);
+            when(minecraftServer.registryAccess()).thenAnswer(ignored -> RegistryHelper.registryAccess());
+            try {
+                var field = net.minecraft.server.MinecraftServer.class.getDeclaredField("SERVER");
+                field.setAccessible(true);
+                field.set(null, minecraftServer);
+            } catch (ReflectiveOperationException exception) {
+                throw new AssertionError("Cannot bind the test server registry", exception);
+            }
+        }
+
         Server instance = mock(withSettings().stubOnly());
 
         when(instance.getItemFactory()).thenAnswer(mock -> CraftItemFactory.instance());
